torch_xla/experimental/pallas_kernels/quantized_matmul_kernel.py

```diff
@@ -181,15 +237,18 @@ def quantized_matmul_int8(
           out_specs=pl.BlockSpec((batch_block_size, out_block_size),
                                  lambda b, o, i: (b, o)),
           scratch_shapes=[
-              pltpu.VMEM((batch_block_size, out_block_size), acc_dtype)
+              pltpu.VMEM((batch_block_size,
+                          out_block_size), acc_dtype) if save_acc else None,
+              pltpu.VMEM((batch_block_size,
+                          in_block_size), jnp.int8) if save_quant else None,
+              pltpu.VMEM(
+                  (batch_block_size, 1), jnp.float32) if save_quant else None,
           ],
-          grid=(padded_bs // batch_block_size,
-                padded_out_features // out_block_size,
-                padded_in_features // in_block_size),
+          grid=(n_bs, n_out, n_in),
       ),
       out_shape=jax.ShapeDtypeStruct((padded_bs, padded_out_features), x.dtype),
       compiler_params=pltpu.TPUCompilerParams(
-          dimension_semantics=("parallel", "parallel", "arbitrary"),
+          dimension_semantics=("parallel", "arbitrary", "arbitrary"),
```

Comment:
out dimension can also be "arbitrary" and executed in any order per https://github.com/jax-ml/jax/blob/b47fa65c32128b98cbc114e9f487478e611181fa/jax/_src/pallas/mosaic/core.py#L75-L77?

Reply:
I've changed this from `parallel` to `arbitrary` because we need to let the kernel know we should store quantization result when `out_idx == 0` and we can retrieve cached quantization when `out_idx != 0`. This logic will not work if out dimension is executed in any order.